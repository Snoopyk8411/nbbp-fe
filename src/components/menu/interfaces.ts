import { Appearance } from 'tools/types/api-catalog-items-types';
import { SubmenuOpenActionType, SubmenuPosition } from './constants';

export interface IMenuProps {
  items: IMenuItemModel[];
  submenuPosition: SubmenuPosition;
}

export interface IMenuLevelProps {
  items: IMenuItemModel[];
  parentItem?: IMenuItemModel;
  submenuOpenActionType: SubmenuOpenActionType;
  submenuPosition: SubmenuPosition;
  containerElement: HTMLElement | null;
  onClose?: () => void;
}

export type MenuItemIdType = string | number;

export interface IMenuItemModel {
  id: MenuItemIdType;
  name: string;
  order: number;
  url: string;
  appearance: Appearance | String;
  children?: IMenuItemModel[];
  svgName: string | undefined;
}
