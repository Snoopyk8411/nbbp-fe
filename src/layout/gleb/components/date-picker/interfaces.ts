export interface IDateData {
  year: number;
  month: number;
  date: number;
}
