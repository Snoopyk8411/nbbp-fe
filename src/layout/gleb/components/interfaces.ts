import { AnyAction, Action, Store, Middleware } from 'redux';

export interface IMockStore<S = any, A extends Action = AnyAction> extends Store<S, A> {
  getActions(): any[];
  clearActions(): void;
}

export type MockStoreEnhanced<S = {}, DispatchExts = {}> = IMockStore<S> & { dispatch: DispatchExts };

export type MockStoreCreator<S = {}, DispatchExts = {}> = (
  state?: S | MockGetState<S>,
) => MockStoreEnhanced<S, DispatchExts>;

export type MockGetState<S = {}> = (actions: AnyAction[]) => S;

/**
 * Create Mock Store returns a function that will create a mock store from a state
 * with the same set of set of middleware applied.
 *
 * @param middlewares The list of middleware to be applied.
 * @template S The type of state to be held by the store.
 * @template DispatchExts The additional Dispatch signatures for the middlewares applied.
 */
declare function createMockStore<S, DispatchExts = {}>(middlewares?: Middleware[]): MockStoreCreator<S, DispatchExts>;

export default createMockStore;

export interface IMockData {
  mediaPage: {
    media: {
      copyright: string;
      date: string;
      explanation: string;
      hdurl: string;
      media_type: string;
      service_version: string;
      title: string;
      url: string;
    };
    error: Error | null;
    selectedDay: any;
    isLoading: boolean;
  };
}
