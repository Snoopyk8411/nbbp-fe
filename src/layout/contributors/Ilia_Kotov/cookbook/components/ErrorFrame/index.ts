import dynamic from 'next/dynamic';

export const ErrorFrame = dynamic(() => import('./ErrorFrame'), { ssr: false });
