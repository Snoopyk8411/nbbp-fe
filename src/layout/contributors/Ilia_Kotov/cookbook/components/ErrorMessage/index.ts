import dynamic from 'next/dynamic';

export const ErrorMessage = dynamic(() => import('./ErrorMessage'), { ssr: false });
