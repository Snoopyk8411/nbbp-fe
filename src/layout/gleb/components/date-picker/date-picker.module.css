.MyDatePicker {
    float: left;
    position: relative;
}

.mdp_input{
    float: left;
    width: 10rem;
    height: 2.5rem;
    overflow: hidden;
}

.mdp_input:focus{
    outline: none;
}
