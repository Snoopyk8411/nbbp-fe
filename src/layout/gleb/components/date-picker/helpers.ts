import { addLeadingZero } from 'tools/formatters/add-leading-zero';

export const getDateStringFromTimestamp = (timestamp: string): string => {
  let dateObject = new Date(timestamp);
  let month = dateObject.getMonth() + 1;
  let date = dateObject.getDate();
  return dateObject.getFullYear() + '-' + addLeadingZero(month) + '-' + addLeadingZero(date);
};
