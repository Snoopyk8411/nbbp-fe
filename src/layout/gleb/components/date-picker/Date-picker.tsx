import { useState, useEffect, createRef, RefObject } from 'react';
import { useDispatch } from 'react-redux';
import { useAppSelector } from 'hooks/use-app-selector';
import { setData } from 'store/gleb/slice';
import { selectNewDateFromDatePicker } from 'store/gleb/selectors';
import { getDateStringFromTimestamp } from './helpers';

import styles from './date-picker.module.css';

type DateProps = {
  onChange: (e: string) => void;
};

export function DatePicker(props: DateProps) {
  const dispatch = useDispatch();
  const date = useAppSelector(selectNewDateFromDatePicker);

  const inputRef: RefObject<HTMLInputElement> = createRef();
  const [isShowDatePicker, setIsShowDatePicker] = useState(false);

  const addBackDrop = (e: MouseEvent) => {
    if (isShowDatePicker && !inputRef.current?.contains(e.target as Node)) {
      setIsShowDatePicker(false);
    }
  };

  const setDateToInput = (timestamp: string) => {
    const dateString = getDateStringFromTimestamp(timestamp);
    if (inputRef.current) inputRef.current.value = dateString;
  };

  useEffect(() => {
    window.addEventListener('click', addBackDrop);
    setDateToInput(date);
    return () => {
      window.removeEventListener('click', addBackDrop);
    };
  }, [isShowDatePicker]);

  const setDateInsideComponent = (dateValue: string) => {
    dispatch(setData(dateValue));
    /** Pass data to parent */
    props.onChange(dateValue);
  };

  const updateDateFromInput = () => {
    const dateValue = inputRef.current?.value;
    if (dateValue && dateValue !== null) {
      setDateInsideComponent(dateValue);
    }
  };

  const handleClick = () => setIsShowDatePicker(true);

  return (
    <div>
      <div className={styles.dp_input} onClick={handleClick}>
        <input type='date' data-testid='inputRef' ref={inputRef} onChange={updateDateFromInput} />
      </div>
    </div>
  );
}
