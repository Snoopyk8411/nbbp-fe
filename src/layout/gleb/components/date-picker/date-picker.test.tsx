import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import createSagaMiddleware from 'redux-saga';
import { Provider } from 'react-redux';
import { DatePicker } from 'layout/gleb/components/date-picker/Date-picker';
import { IMockStore } from 'layout/gleb/components/interfaces';

let mockStore: IMockStore;
const sagaMiddleware = createSagaMiddleware();
const mockStoreConf = configureStore([sagaMiddleware]);

const mockData = {
  picturePage: {
    picture: {
      copyright: '',
      date: '2021-11-07',
      explanation: "To some it looks like a cat's eye.",
      hdurl: 'https://apod.nasa.gov/apod/image/2111/CatsEye_HubblePohl_1278.jpg',
      media_type: 'image',
      service_version: 'v1',
      title: "The Cat's Eye Nebula in Optical and X-ray",
      url: 'https://apod.nasa.gov/apod/image/2111/CatsEye_HubblePohl_960.jpg',
    },
    error: null,
    selectedDay: 1637010000000,
  },
};
type DateProps = {
  onChange: (e: string) => void;
};

describe('DatePicker Component', () => {
  let props: DateProps;
  beforeEach(() => {
    mockStore = mockStoreConf(mockData);
  });

  it('DatePicker snapshot', () => {
    const component = renderer.create(
      <Provider store={mockStore}>
        <DatePicker {...props} />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
