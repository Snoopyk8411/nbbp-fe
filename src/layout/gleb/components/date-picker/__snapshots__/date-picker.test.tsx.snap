// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Component DatePicker snapshot 1`] = `
<div
  className="MyDatePicker"
>
  <div
    className="mdp_input"
    onClick={[Function]}
  >
    <input
      data-testid="inputRef"
      onChange={[Function]}
      type="date"
    />
  </div>
</div>
`;
